### ISSUE TITLE
Notebooks - Egeria servers do not start (2.3/master)

### ISSUE THREAD

**Issue by planetf1** [odpi/egeria#3698]
When running the egeria notebooks locally, or in containers, whilst the config notebook & start notebook complete successfully, the servers are not started

**Comment by planetf1** [comment]
The cause of this is that in the environment-check.ipynb notebook the function `checkServerActive()` makes a rest call
to something like `https://localhost:9444/open-metadata/platform-services/users/garygeeke/server-platform/servers/cocoMDS2/status` to determine if a server is active before starting it

The *current* response when a server is not running is:
```
{'class': 'ServerStatusResponse', 'relatedHTTPCode': 404, 'exceptionClassName': 'org.odpi.openmetadata.frameworks.connectors.ffdc.InvalidParameterException', 'actionDescription': 'getServerStartTimeFromPlatform', 'exceptionErrorMessage': 'OMAG-MULTI-TENANT-404-001 The OMAG Server cocoMDS2 is not available to service a request from user garygeeke', 'exceptionErrorMessageId': 'OMAG-MULTI-TENANT-404-001', 'exceptionErrorMessageParameters': ['cocoMDS2', 'garygeeke'], 'exceptionSystemAction': 'The system is unable to process the request because the server is not running.', 'exceptionUserAction': 'Retry the request when the OMAG Server is available.', 'exceptionProperties': {'serverName': 'cocoMDS2', 'parameterName': 'serverName'}}
```
which the function reports as `... cocoMDS2 is down - needs to be started`

however the exit from this function is set as
```
        serverStatus = response.json().get('active')
```
This resolves to 'None' with the above result.

Whilst a simple fix is to ensure False is returned, in fact this is an error case here -- that function should be reporting `is down - unable to check server configuration`. Furthermore the API call shouldn't get back an 'InvalidParameterException' - this seems to be a *change in behaviour* 

Looking at the actual call - in fact that API is returning status code 200 - so the response is very confused.

Checking for recent changes ...

**Comment by planetf1** [comment]
For clarity, the full response from the status query of a server which is down is actually:
```
$ http --verify no --pretty=format GET https://localhost:9444/open-metadata/platform-services/users/garygeeke/server-platform/servers/cocoMDS2/status
HTTP/1.1 200
Connection: keep-alive
Content-Type: application/json
Date: Mon, 28 Sep 2020 15:38:41 GMT
Keep-Alive: timeout=60
Transfer-Encoding: chunked

{
    "actionDescription": "getServerStartTimeFromPlatform",
    "class": "ServerStatusResponse",
    "exceptionClassName": "org.odpi.openmetadata.frameworks.connectors.ffdc.InvalidParameterException",
    "exceptionErrorMessage": "OMAG-MULTI-TENANT-404-001 The OMAG Server cocoMDS2 is not available to service a request from user garygeeke",
    "exceptionErrorMessageId": "OMAG-MULTI-TENANT-404-001",
    "exceptionErrorMessageParameters": [
        "cocoMDS2",
        "garygeeke"
    ],
    "exceptionProperties": {
        "parameterName": "serverName",
        "serverName": "cocoMDS2"
    },
    "exceptionSystemAction": "The system is unable to process the request because the server is not running.",
    "exceptionUserAction": "Retry the request when the OMAG Server is available.",
    "relatedHTTPCode": 404
}

```

**Comment by planetf1** [comment]
^^ cc: @grahamwallis @davidradl @mandy-chessell see regression in 2.3 (and 2.4/master). Trying to pin down origin of breakage. maybe refactoring or view service work...

**Comment by planetf1** [comment]
On closer inspection the only difference with earlier releases, like 2.1 is that in that release we had:
```
    "serverName": "cocoMDS2"
```
as the very last element of the response - that is no longer present

**Comment by planetf1** [comment]
Example from running the notebook - the first server  on 19443 is backlevel (from 2.2) whilst on 9444 we have the current master code
```
        Checking OMAG Server cocoMDS6
           ... cocoMDS6 is configured
True
https://localhost:19443/open-metadata/platform-services/users/garygeeke/server-platform/servers/cocoMDS6/status
200
{'class': 'ServerStatusResponse', 'relatedHTTPCode': 404, 'exceptionClassName': 'org.odpi.openmetadata.frameworks.connectors.ffdc.InvalidParameterException', 'actionDescription': 'getServerStartTimeFromPlatform', 'exceptionErrorMessage': 'OMAG-MULTI-TENANT-404-001 The OMAG Server cocoMDS6 is not available to service a request from user garygeeke', 'exceptionErrorMessageId': 'OMAG-MULTI-TENANT-404-001', 'exceptionErrorMessageParameters': ['cocoMDS6', 'garygeeke'], 'exceptionSystemAction': 'The system is unable to process the request because the server is not running.', 'exceptionUserAction': 'Retry the request when the OMAG Server is available.', 'exceptionProperties': {'serverName': 'cocoMDS6', 'parameterName': 'serverName'}, 'serverName': 'cocoMDS6', 'active': False}
           ... cocoMDS6 is down - needs to be started
False
False
                Starting server cocoMDS6 ...
starting
    Data Lake Platform is active
        Checking OMAG Server cocoMDS1
           ... cocoMDS1 is configured
True
https://localhost:9444/open-metadata/platform-services/users/garygeeke/server-platform/servers/cocoMDS1/status
200
{'class': 'ServerStatusResponse', 'relatedHTTPCode': 200, 'serverStatus': {'serverName': 'cocoMDS1', 'isActive': True, 'serverStartTime': '2020-09-28T15:14:46.893+00:00', 'serverHistory': [{'startTime': '2020-09-28T13:52:19.174+00:00', 'endTime': '2020-09-28T15:14:16.431+00:00'}, {'startTime': '2020-09-28T15:14:17.189+00:00', 'endTime': '2020-09-28T15:14:46.464+00:00'}]}}
           ... cocoMDS1 is down - needs to be started
None
None
```

**Comment by planetf1** [comment]
the 200 vs 404 was down to my experimenting with servers - in fact the core difference that affects the logic is that in master we've lost the parameter values in the exception ie:

```
'exceptionProperties': {'serverName': 'cocoMDS4', 'parameterName': 'serverName'}
```
which should be
```
{'serverName': 'cocoMDS6', 'parameterName': 'serverName'}, 'serverName': 'cocoMDS6', 'active': False}
```

and the python code is looking for the value of 'active' to return true or false - with the failure to start stemming from there

**Comment by planetf1** [comment]
@davidradl @grahamwallis  Was this area changed as part of integrating the view services
Or @davidradl @immmus any of the refactoring (though that was mostly subject area?)
cc: @mandy-chessell 

This is potentially a release blocker.

**Comment by immmus** [comment]
> @davidradl @grahamwallis Was this area changed as part of integrating the view services
> Or @davidradl @immmus any of the refactoring (though that was mostly subject area?)
> cc: @mandy-chessell
> 
> This is potentially a release blocker.

Yes I made mostly subject area changes.

**Comment by grahamwallis** [comment]
@planetf1 - Sorry I am almost 100% certain that this is my fault. I changed ServerStatusResponse when I wrote the Platform Services client. I should have got this cleared in a proposal issue but I think it got buried in amongst the many changes I made to facilitate Dino and what probably started out as an experiment became a committed change.

**Comment by grahamwallis** [comment]
Looking back at my journal re the reasons for this change I noted at the time that "we want the server start/end history  (requested during feedback from Dino demo on the ODPi call on Thursday 20th August and in discussion with Mandy) so I bit the bullet and have changed ServerStatusResponse and included a new method in the PlatformServicesClient. This means that a ServerOverview object contains current status, so when a server is clicked it will refresh and the solid/dotted line showing whether it is running will be updated to reflect the current state."

If possible I'd like to keep the ServerStatus object if it is possible for other code that uses the ServerStatusResponse to be updated to cope with my unsolicited change. Otherwise I will revert it and find another way to do the Dino status updates.

**Comment by mandy-chessell** [comment]
There is currently a method in the platform services to retrieve the server status and history - no need to make any changes

**Comment by grahamwallis** [comment]
There is - i.e. the platform services change I think was complete - but for something that calls the REST API directly and parses the REST response I think I have moved the status field 'down' one level into the ServerStatus object, which will  catch any such users by surprise and I should have raised a proper proposal I think and tried to maintain back compatibility. If the only users are our notebooks and test code I might get away with it but I acknowledge that I may need to revert it.

**Comment by planetf1** [comment]
Thanks @grahamwallis - however in the 'new' version I don't see state reported at all - we have a top level 200 response, but an embedded 404 - since our virtual server isn't running. But the query is to retrieve state, which previously was returned. I don't now see this anywhere in the response?

Of course there's no problem changing the notebook -- but the real questions are
 a) what is the effect on others
 b) what should the correct behaviour be for this status query

**Comment by grahamwallis** [comment]
@planetf1 - Nigel, are you satisfied that this is now working and fixed?

**Comment by planetf1** [comment]
yes - we can close . It was fixed in PR #3706
